accesscontroltool-content-package-2.7.2.zip blocks content distribution queue in AEMaaCS

The deployment fails and the following error can be observed in the publish logs:

```
15.03.2021 17:35:22.248 [cm-p29008-e106842-aem-publish-7c69f78f56-rb9wv] *INFO* [Queue Processor for Subscriber agent publishSubscriber] org.apache.sling.distribution.journal.bookkeeper.ContentPackageExtractor Content package received at /etc/packages/Netcentric/accesscontroltool-content-package-2.7.2.zip. Starting import.
15.03.2021 17:35:22.293 [cm-p29008-e106842-aem-publish-7c69f78f56-rb9wv] *ERROR* [Queue Processor for Subscriber agent publishSubscriber] org.apache.jackrabbit.vault.fs.impl.io.GenericArtifactHandler Error while parsing /jcr_root/_rep_policy.xml: {}
org.xml.sax.SAXException: javax.jcr.AccessDeniedException: Access denied.
	at org.apache.jackrabbit.vault.fs.impl.io.DocViewSAXImporter.endElement(DocViewSAXImporter.java:1245) [org.apache.jackrabbit.vault:3.4.0]
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.endElement(AbstractSAXParser.java:610)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanEndElement(XMLDocumentFragmentScannerImpl.java:1718)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:2883)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:112)
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:534)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:888)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:824)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
	at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1216)
	at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:635)
	at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl.parse(SAXParserImpl.java:324)
	at org.apache.jackrabbit.vault.fs.impl.io.GenericArtifactHandler.accept(GenericArtifactHandler.java:100) [org.apache.jackrabbit.vault:3.4.0]
	at org.apache.jackrabbit.vault.fs.io.Importer.commit(Importer.java:896) [org.apache.jackrabbit.vault:3.4.0]
	at org.apache.jackrabbit.vault.fs.io.Importer.commit(Importer.java:799) [org.apache.jackrabbit.vault:3.4.0]
	at org.apache.jackrabbit.vault.fs.io.Importer.commit(Importer.java:839) [org.apache.jackrabbit.vault:3.4.0]
	at org.apache.jackrabbit.vault.fs.io.Importer.run(Importer.java:440) [org.apache.jackrabbit.vault:3.4.0]
	at org.apache.jackrabbit.vault.packaging.impl.ZipVaultPackage.extract(ZipVaultPackage.java:232) [org.apache.jackrabbit.vault:3.4.0]
	at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:401) [org.apache.jackrabbit.vault:3.4.0]
	at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:360) [org.apache.jackrabbit.vault:3.4.0]
	at org.apache.jackrabbit.vault.packaging.impl.JcrPackageImpl.extract(JcrPackageImpl.java:346) [org.apache.jackrabbit.vault:3.4.0]
	at org.apache.sling.distribution.journal.bookkeeper.ContentPackageExtractor.installPackage(ContentPackageExtractor.java:101) [org.apache.sling.distribution.journal:0.2.0.T202009251421-0284693]
	at org.apache.sling.distribution.journal.bookkeeper.ContentPackageExtractor.installPackage(ContentPackageExtractor.java:93) [org.apache.sling.distribution.journal:0.2.0.T202009251421-0284693]
	at org.apache.sling.distribution.journal.bookkeeper.ContentPackageExtractor.handle(ContentPackageExtractor.java:72) [org.apache.sling.distribution.journal:0.2.0.T202009251421-0284693]
	at org.apache.sling.distribution.journal.bookkeeper.PackageHandler.installAddPackage(PackageHandler.java:79) [org.apache.sling.distribution.journal:0.2.0.T202009251421-0284693]
	at org.apache.sling.distribution.journal.bookkeeper.PackageHandler.apply(PackageHandler.java:61) [org.apache.sling.distribution.journal:0.2.0.T202009251421-0284693]
	at org.apache.sling.distribution.journal.bookkeeper.BookKeeper.importPackage(BookKeeper.java:148) [org.apache.sling.distribution.journal:0.2.0.T202009251421-0284693]
	at org.apache.sling.distribution.journal.impl.subscriber.DistributionSubscriber.processQueueItem(DistributionSubscriber.java:339) [org.apache.sling.distribution.journal:0.2.0.T202009251421-0284693]
	at org.apache.sling.distribution.journal.impl.subscriber.DistributionSubscriber.fetchAndProcessQueueItem(DistributionSubscriber.java:296) [org.apache.sling.distribution.journal:0.2.0.T202009251421-0284693]
	at org.apache.sling.distribution.journal.impl.subscriber.DistributionSubscriber.processQueue(DistributionSubscriber.java:275) [org.apache.sling.distribution.journal:0.2.0.T202009251421-0284693]
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: javax.jcr.AccessDeniedException: Access denied.
	at org.apache.jackrabbit.oak.spi.security.authorization.accesscontrol.AbstractAccessControlManager.checkPermissions(AbstractAccessControlManager.java:208) [org.apache.jackrabbit.oak-security-spi:1.37.0.R1885266]
	at org.apache.jackrabbit.oak.spi.security.authorization.accesscontrol.AbstractAccessControlManager.getTree(AbstractAccessControlManager.java:168) [org.apache.jackrabbit.oak-security-spi:1.37.0.R1885266]
	at org.apache.jackrabbit.oak.spi.security.authorization.principalbased.impl.PrincipalBasedAccessControlManager.getPolicies(PrincipalBasedAccessControlManager.java:161) [org.apache.jackrabbit.oak-authorization-principalbased:1.37.0.R1885266]
	at org.apache.jackrabbit.oak.security.authorization.composite.CompositeAccessControlManager.getPolicies(CompositeAccessControlManager.java:82) [org.apache.jackrabbit.oak-core:1.37.0.R1885266]
	at org.apache.jackrabbit.oak.jcr.delegate.AccessControlManagerDelegator$5.perform(AccessControlManagerDelegator.java:97) [org.apache.jackrabbit.oak-jcr:1.37.0.R1885266]
	at org.apache.jackrabbit.oak.jcr.delegate.AccessControlManagerDelegator$5.perform(AccessControlManagerDelegator.java:93) [org.apache.jackrabbit.oak-jcr:1.37.0.R1885266]
	at org.apache.jackrabbit.oak.jcr.delegate.SessionDelegate.perform(SessionDelegate.java:209) [org.apache.jackrabbit.oak-jcr:1.37.0.R1885266]
	at org.apache.jackrabbit.oak.jcr.delegate.AccessControlManagerDelegator.getPolicies(AccessControlManagerDelegator.java:93) [org.apache.jackrabbit.oak-jcr:1.37.0.R1885266]
	at org.apache.jackrabbit.oak.jcr.delegate.JackrabbitAccessControlManagerDelegator.getPolicies(JackrabbitAccessControlManagerDelegator.java:141) [org.apache.jackrabbit.oak-jcr:1.37.0.R1885266]
	at org.apache.jackrabbit.vault.fs.impl.io.JackrabbitACLImporter$ImportedPolicy.getPolicy(JackrabbitACLImporter.java:182) [org.apache.jackrabbit.vault:3.4.0]
	at org.apache.jackrabbit.vault.fs.impl.io.JackrabbitACLImporter$ImportedAcList.apply(JackrabbitACLImporter.java:267) [org.apache.jackrabbit.vault:3.4.0]
	at org.apache.jackrabbit.vault.fs.impl.io.JackrabbitACLImporter.close(JackrabbitACLImporter.java:154) [org.apache.jackrabbit.vault:3.4.0]
	at org.apache.jackrabbit.vault.fs.impl.io.DocViewSAXImporter.endElement(DocViewSAXImporter.java:1189) [org.apache.jackrabbit.vault:3.4.0]
	... 31 common frames omitted

```

--- a/README.md
+++ b/README.md
@@ -36,6 +36,14 @@ The [content package](https://jackrabbit.apache.org/filevault) is available from
     <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
     <artifactId>accesscontroltool-package</artifactId>
 ```
+(for AEM Classic/On Premise) or
+
+```
+    <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
+    <artifactId>accesscontroltool-package</artifactId>
+    <classifier>cloud</classifier>
+```
+(for AEM as a Cloud Service)
 
 Install it afterwards e.g. via AEM's package manager.
 

--- a/accesscontroltool-apps-package/pom.xml
+++ b/accesscontroltool-apps-package/pom.xml
@@ -15,7 +15,7 @@
     <parent>
         <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
         <artifactId>accesscontroltool</artifactId>
-        <version>2.7.3-SNAPSHOT</version>
+        <version>2.8.0-SNAPSHOT</version>
     </parent>
 
     <!-- ====================================================================== -->

--- a/accesscontroltool-bundle/pom.xml
+++ b/accesscontroltool-bundle/pom.xml
@@ -11,7 +11,7 @@
     <parent>
         <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
         <artifactId>accesscontroltool</artifactId>
-        <version>2.7.3-SNAPSHOT</version>
+        <version>2.8.0-SNAPSHOT</version>
     </parent>
 
     <!-- ====================================================================== -->

--- a/accesscontroltool-content-package/pom.xml
+++ b/accesscontroltool-content-package/pom.xml
@@ -15,7 +15,7 @@
     <parent>
         <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
         <artifactId>accesscontroltool</artifactId>
-        <version>2.7.3-SNAPSHOT</version>
+        <version>2.8.0-SNAPSHOT</version>
     </parent>
 
     <!-- ====================================================================== -->

--- a/accesscontroltool-exampleconfig-package/pom.xml
+++ b/accesscontroltool-exampleconfig-package/pom.xml
@@ -15,7 +15,7 @@
     <parent>
         <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
         <artifactId>accesscontroltool</artifactId>
-        <version>2.7.3-SNAPSHOT</version>
+        <version>2.8.0-SNAPSHOT</version>
     </parent>
 
     <!-- ====================================================================== -->

--- a/accesscontroltool-oakindex-package/pom.xml
+++ b/accesscontroltool-oakindex-package/pom.xml
@@ -15,7 +15,7 @@
     <parent>
         <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
         <artifactId>accesscontroltool</artifactId>
-        <version>2.7.3-SNAPSHOT</version>
+        <version>2.8.0-SNAPSHOT</version>
     </parent>
 
     <!-- ====================================================================== -->

--- a/accesscontroltool-package/pom.xml
+++ b/accesscontroltool-package/pom.xml
@@ -170,6 +170,7 @@
                             <!-- must set explicitly due to https://issues.apache.org/jira/browse/JCRVLT-498 -->
                             <workDirectory>${project.build.directory}/vault-work-cloud</workDirectory>
                             <classifier>cloud</classifier>
+                            <failOnDuplicateEntries>false</failOnDuplicateEntries>
                         </configuration>
                     </execution>
                     <execution>

--- a/accesscontroltool-package/pom.xml
+++ b/accesscontroltool-package/pom.xml
@@ -15,7 +15,7 @@
     <parent>
         <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
         <artifactId>accesscontroltool</artifactId>
-        <version>2.7.3-SNAPSHOT</version>
+        <version>2.8.0-SNAPSHOT</version>
     </parent>
 
     <!-- ====================================================================== -->

--- a/accesscontroltool-package/pom.xml
+++ b/accesscontroltool-package/pom.xml
@@ -77,39 +77,101 @@
                     </execution>
                 </executions>
             </plugin>
+            <!-- for "cloud" package use repoinit instead of actool-content-package -->
+            <plugin>
+                <artifactId>maven-resources-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>copy-cloud-package-resources</id>
+                        <!-- here the phase you need -->
+                        <phase>process-resources</phase>
+                        <goals>
+                            <goal>copy-resources</goal>
+                        </goals>
+                        <configuration>
+                            <outputDirectory>${project.build.directory}/vault-work-cloud/jcr_root</outputDirectory>
+                            <resources>
+                              <resource>
+                                <directory>src/main/jcr_root-cloud</directory>
+                                <filtering>false</filtering>
+                              </resource>
+                            </resources>
+                        </configuration> 
+                    </execution>
+                </executions>
+            </plugin>
+            <!-- generate two packages: one for cloud (classifier "cloud") and one for classic (no classifier) -->
             <plugin>
                 <groupId>org.apache.jackrabbit</groupId>
                 <artifactId>filevault-package-maven-plugin</artifactId>
                 <configuration>
-                    <properties>
-                        <!-- install the two bundles and service user config synchronously in order to be callable from within 
-                              same complete package (if yaml file containing package declares the dependency to it) -->
-                        <installPathRegex>/apps/netcentric/actool/(config|install)/.*</installPathRegex>
-                        <waitForOsgiEventsQuietInSec>5</waitForOsgiEventsQuietInSec>
-                    </properties>
                     <embeddedTarget>/apps/netcentric/actool/install</embeddedTarget>
-                    <embeddeds>
-                        <embedded>
-                            <artifactId>accesscontroltool-bundle</artifactId>
-                        </embedded>
-                        <embedded>
-                            <artifactId>accesscontroltool-startuphook-bundle</artifactId>
-                            <target>/apps/netcentric/actool/install/27</target>
-                        </embedded>
-                    </embeddeds>
-                    <subPackages>
-                        <subPackage>
-                            <artifactId>accesscontroltool-apps-package</artifactId>
-                        </subPackage>
-                        <subPackage>
-                            <artifactId>accesscontroltool-content-package</artifactId>
-                        </subPackage>
-                    </subPackages>
                     <packageType>container</packageType>
                     <!-- all sub packages have been validated already -->
                     <skipSubPackageValidation>true</skipSubPackageValidation>
                 </configuration>
                 <executions>
+                    <execution>
+                        <id>default-generate-metadata</id>
+                        <configuration>
+                            <properties>
+                                <!-- install the two bundles and service user config synchronously in order to be callable from within 
+                                      same complete package (if yaml file containing package declares the dependency to it) -->
+                                <installPathRegex>/apps/netcentric/actool/(config|install)/.*</installPathRegex>
+                                <waitForOsgiEventsQuietInSec>5</waitForOsgiEventsQuietInSec>
+                            </properties>
+                            <embeddeds>
+                                <embedded>
+                                    <artifactId>accesscontroltool-bundle</artifactId>
+                                </embedded>
+                                <embedded>
+                                    <artifactId>accesscontroltool-startuphook-bundle</artifactId>
+                                    <target>/apps/netcentric/actool/install/27</target>
+                                </embedded>
+                            </embeddeds>
+                            <subPackages>
+                                <subPackage>
+                                    <artifactId>accesscontroltool-apps-package</artifactId>
+                                </subPackage>
+                                <subPackage>
+                                    <artifactId>accesscontroltool-content-package</artifactId>
+                                </subPackage>
+                            </subPackages>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>cloud-generate-metadata</id>
+                        <goals>
+                            <goal>generate-metadata</goal>
+                        </goals>
+                        <configuration>
+                            <!-- everything via embed, no subpackages, no accesscontroltool-content-package -->
+                            <embeddeds>
+                                <embedded>
+                                    <artifactId>accesscontroltool-bundle</artifactId>
+                                </embedded>
+                                <embedded>
+                                    <artifactId>accesscontroltool-apps-package</artifactId>
+                                </embedded>
+                                <embedded>
+                                    <artifactId>accesscontroltool-startuphook-bundle</artifactId>
+                                    <target>/apps/netcentric/actool/install/27</target>
+                                </embedded>
+                            </embeddeds>
+                            <classifier>cloud</classifier>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>cloud-package</id>
+                        <goals>
+                            <goal>package</goal>
+                        </goals>
+                        <configuration>
+                            <!-- must set explicitly due to https://issues.apache.org/jira/browse/JCRVLT-498 -->
+                            <workDirectory>${project.build.directory}/vault-work-cloud</workDirectory>
+                            <classifier>cloud</classifier>
+                        </configuration>
+                    </execution>
                     <execution>
                         <id>default-analyze-classes</id>
                         <phase>invalid</phase>

--- a/accesscontroltool-package/src/main/jcr_root-cloud/apps/netcentric/actool/config/org.apache.sling.jcr.repoinit.RepositoryInitializer~actool.config
+++ b/accesscontroltool-package/src/main/jcr_root-cloud/apps/netcentric/actool/config/org.apache.sling.jcr.repoinit.RepositoryInitializer~actool.config
@@ -1,8 +1,6 @@
-scripts=["create service user actool-service with path system/actool
+scripts=["create service user actool-service with forced path system/cq:services/actool
 set principal ACL for actool-service
     allow jcr:all on /
-end
-set repository ACL for actool-service
-    allow jcr:all
+    allow jcr:all on :repository 
 end
 "]

--- a/accesscontroltool-package/src/main/jcr_root-cloud/apps/netcentric/actool/config/org.apache.sling.jcr.repoinit.RepositoryInitializer~actool.config
+++ b/accesscontroltool-package/src/main/jcr_root-cloud/apps/netcentric/actool/config/org.apache.sling.jcr.repoinit.RepositoryInitializer~actool.config
@@ -1,0 +1,8 @@
+scripts=["create service user actool-service with path system/actool
+set principal ACL for actool-service
+    allow jcr:all on /
+end
+set repository ACL for actool-service
+    allow jcr:all
+end
+"]

--- a/accesscontroltool-package/src/main/jcr_root-cloud/apps/netcentric/actool/config/org.apache.sling.serviceusermapping.impl.ServiceUserMapperImpl.amended-ncActoolSystemUser.config
+++ b/accesscontroltool-package/src/main/jcr_root-cloud/apps/netcentric/actool/config/org.apache.sling.serviceusermapping.impl.ServiceUserMapperImpl.amended-ncActoolSystemUser.config
@@ -1,0 +1,1 @@
+user.mapping=["biz.netcentric.cq.tools.accesscontroltool.bundle\=[actool-service]","biz.netcentric.cq.tools.accesscontroltool.startuphook.bundle\=[actool-service]"]

--- a/accesscontroltool-startuphook-bundle/pom.xml
+++ b/accesscontroltool-startuphook-bundle/pom.xml
@@ -11,7 +11,7 @@
     <parent>
         <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
         <artifactId>accesscontroltool</artifactId>
-        <version>2.7.3-SNAPSHOT</version>
+        <version>2.8.0-SNAPSHOT</version>
     </parent>
 
     <!-- ====================================================================== -->

--- a/pom.xml
+++ b/pom.xml
@@ -11,7 +11,7 @@
 
     <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
     <artifactId>accesscontroltool</artifactId>
-    <version>2.7.3-SNAPSHOT</version>
+    <version>2.8.0-SNAPSHOT</version>
     <packaging>pom</packaging>
 
     <name>Access Control Tool</name>

--- a/target-osgi-environment/maximum-environment/pom.xml
+++ b/target-osgi-environment/maximum-environment/pom.xml
@@ -21,7 +21,7 @@
         <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
         <artifactId>target-osgi-environment</artifactId>
         <relativePath>../pom.xml</relativePath>
-        <version>2.7.3-SNAPSHOT</version>
+        <version>2.8.0-SNAPSHOT</version>
     </parent>
 
     <artifactId>maximum-environment</artifactId>

--- a/target-osgi-environment/minimum-environment/pom.xml
+++ b/target-osgi-environment/minimum-environment/pom.xml
@@ -21,7 +21,7 @@
         <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
         <artifactId>target-osgi-environment</artifactId>
         <relativePath>../pom.xml</relativePath>
-        <version>2.7.3-SNAPSHOT</version>
+        <version>2.8.0-SNAPSHOT</version>
     </parent>
 
     <artifactId>minimum-environment</artifactId>

--- a/target-osgi-environment/pom.xml
+++ b/target-osgi-environment/pom.xml
@@ -20,7 +20,7 @@
         <groupId>biz.netcentric.cq.tools.accesscontroltool</groupId>
         <artifactId>accesscontroltool</artifactId>
         <relativePath>../pom.xml</relativePath>
-        <version>2.7.3-SNAPSHOT</version>
+        <version>2.8.0-SNAPSHOT</version>
     </parent>
 
     <artifactId>target-osgi-environment</artifactId>

